### ISSUE TITLE
Efficient way to get Tracker.count

### ISSUE THREAD

**Issue by michelson** [phoenixframework/phoenix_pubsub#65]
Hi, I've been playing with `phoenix_pubsub` with the tracker feature,  I've built an http endpoint to show how many tracked clients the system holds, using this:

    `size = length( Phoenix.Tracker.list(My.Tracker, "bucket"))`

when system has more than 10k tracker in the registry the length function takes for ever to process. is there any way to call a count function over the `Phoenix.PubSub.PG2` registry in a more efficient way ?

**Comment by chrismccord** [comment]
`Phoenix.Tracker.list(My.Tracker, "bucket")` actually returns a map so `length` would throw an argument error. Are you sure this is the code path that's being hit on the slowdown? It definitely would raise. Enum.count would defer to capsize, which should be constant time for maps

**Comment by michelson** [comment]
Hi @chrismccord , thanks for your reply, 
It seems I get the same results for `length` and `Enum.count`, both returns the size correctly , no exceptions, 
I'm doing some load testing on the same machine were i'm doing the count. if this `count` is not the bottleneck I guess the slowness could be because the load (20k concurrent reqs). by the way I'm receiving a lot of timeouts from Phoenix pubsub. but memory & cpu are way below 100% 


```
14:58:13.545 [error] #PID<0.20804.4> running MyApp terminated
Server: ip:4000 (http)
Request: GET /api/d/xxx.json
** (exit) exited in: GenServer.call(MyApp.PubSub.Local0, {:monitor, #PID<0.20804.4>, []}, 5000)
    ** (EXIT) time out
```

**Comment by michelson** [comment]
Hi, doing some testing I've been able to make the count from the console on the same machine and from a second node , and the response is fast from both, so I guess the problem has to be on the plug endpoint.

The timeouts on PubSub also occurs when the load tests ends, so I believe the timeout happens when connection is interrupted.

do you have any idea why the server is too slow in my /connected endpoint ? should I put this on another process as an strategy to improve performance?

best regards

**Comment by chrismccord** [comment]
We don't have enough information to say what the bottleneck is as it's not clear what your app is doing and what you are measuring. What version of phoenix_pubsub are you using. You only showed `Local0` as the timeout. Is this is the only server name you see timing out? If so, you have a pool size of 1, which you should increase. By default as of a recent release we default to the number of schedulers. Try setting the following in your config:

    config :my_app, MyApp.Endpoint, pubsub: [name: MyApp.PubSub, pool_size: 10, ...]

**Comment by michelson** [comment]
well , I've a mix application which is an umbrella app. 

**app1:**
 + cowboy, 
 + plug 
 + app2

**app2:**
 + pub sub app
 + tracker
 + redix commands

app1 includes app 2, app1 has an endpoint which does a long polling, it starts a pubsub on each request waiting to receive messages for 60_000, my tests are 20k connections with 20k concurrent users , expecting the 100% to be completed in ~70s at most

In the app2 I have the following:

```elixir
  def start(_type, _args) do
    import Supervisor.Spec, warn: false

    redis_host = :os.getenv("REDIS_HOST") || "localhost"

    children = [
      worker(Redix, [[host: redis_host, port: 6379], 
        [name: :redix]]),

      supervisor(Phoenix.PubSub.PG2, [MyApp.PubSub, [pool_size: 20]]),
      
      worker(MyApp.Tracker, 
        [[name: MyApp.Tracker, pubsub_server: MyApp.PubSub]])
    ]
    opts = [strategy: :one_for_one, name: MyApp.Supervisor]
    Supervisor.start_link(children, opts)
  end
```

after setup the pool size for PubSub I'm receiving timeouts for MyApp.Tracker, 
should I put a pool size for Tracker too ?
should I put some pooling on cowboy, (and redix) server too ?

thanks for your feedback @chrismccord , very thankful

**Comment by michelson** [comment]
Hi! I've tunned some settings in cowboy webserver and now it's working fine, I've increased the `max_connections` & `acceptors`, like this:

`Plug.Adapters.Cowboy.child_spec(:http, NotifyPolling, [], 
        [port: 4000, max_connections: 500000, acceptors: 10000 ]),`

I guess I've receiving the Timeouts on Tracker because Cowboy Plug `max_connections` defaults to 16384 and the timeout is 5000ms, so Tracker hits timeout from Cowboy ?

Anyhow, I don't know which is the proper configuration for this, but now the `/connected` endpoint respond fast and the Timeouts on Tracker are gone.

It seems Cowboy counts http request for the max_connection limit, but not websocket, as I've read.

thanks for your help @chrismccord , have a happy holidays
